docs/backend/testing/running-integration-tests-with-routing-middleware-enabled.md

```diff
@@ -0,0 +1,105 @@
+---
+sidebar_position: 8
+---
+
+# Running integration tests with routing middleware enabled
+
+Sometimes it will be necessary to write server integration tests with routing middleware enabled. For example, we want to test calling an API endpoint with an unauthorized user, and ensure that we receive a 403 response. Here's how to go about it.
+
+## Test name and location
+
+Your test will live in `pkg/handlers/routing`, in a subdirectory corresponding to the API you're testing, `{api_name}api_test`. Tests of the `internal` API live in `pkg/handlers/routing/internalapi_test`.
+
+The name of the file should correspond to the handler that the endpoint calls, `{handler}_test.go`. `pkg/handlers/internalapi/ppm_shipment.go` would have a corresponding test file in `pkg/handlers/routing/internalapi_test/ppm_shipment_test.go`.
+
+Generally, the test name and location should be
+```pkg/handlers/routing/{api_name}api_test/{handler_name}_test.go```
+
+## Test structure
+
+In general, here's how to set up the file created above.
+
+Include the test as part of the `{api_name}api_test` package, and any imports, as usual.
+
+For each request type, write a function whose receiver is the API's test suite. For example, tests of updating a pro-gear weight ticket in the internal API's suite would look like:
+
+```
+func (suite *InternalAPISuite) TestUpdateProgearWeightTicket() {
+  ...
+}
+```
+
+For each test, call `Run` on the suite, with a description of the test, and the test function.
+
+```
+  suite.Run("Unauthorized progear weight ticket update by another service member", func() { ... })
+```
+
+The body of the test function will likely contain some data set up -- using factories to build a pro-gear weight ticket and service member, for example -- which we can use to build the endpoint, body, and request, and get the response.
+### Endpoint
+
```

Comment:
**thought**: Your example shows this, but I think it would be good to explicitly call it out. When setting up the endpoints for these types of tests, you need to prefix the endpoint path with `/internal`, `/ghc/v1`, `/prime/v1`, etc. This is a difference from the endpoint paths that we usually set up in our handler server and integration tests that live next to the handler files because those omit that part of the endpoint.

Reply:
Added, nice suggestion.